src/codegen/ast_interpreter.cpp

```diff
@@ -609,6 +630,21 @@ Value ASTInterpreter::visit_invoke(AST_Invoke* node) {
         v = visit_stmt(node->stmt);
         next_block = node->normal_dest;
     } catch (ExcInfo e) {
+
+        if (threading::ThreadStateInternal::getUnwindState() == UNWIND_STATE_NORMAL) {
```

Comment:
alas, this doesn't work.  passing `None` as the third arg (as in  `raise exc, tp, None`) should not be treated as a re-raise, but we can't determine that by looking at the AST_stmt.